src/async/deserializeAsync.ts

```diff
@@ -62,10 +73,11 @@ function createTsonDeserializer(opts: TsonAsyncOptions) {
 		iterable: TsonDeserializeIterable,
 		parseOptions: TsonParseAsyncOptions,
 	) => {
-		const cache = new Map<
+		const controllers = new Map<
 			TsonAsyncIndex,
 			ReadableStreamDefaultController<unknown>
 		>();
+		const cache = new Map<TsonAsyncIndex, unknown>();
```

Comment:
I'd like this to be a `WeakMap` so we don't cache all resolved promises etc but weakmaps can't be used with a number as the index